lib/bike_brigade/importers/mailchimp_importer.ex

```diff
@@ -7,83 +7,77 @@ defmodule BikeBrigade.Importers.MailchimpImporter do
   alias BikeBrigade.Importers.Importer
   alias BikeBrigade.Riders
   alias BikeBrigade.Riders.Rider
+  alias BikeBrigade.Messaging.SlackWebhook
 
   @count 100
   @mailchimp "mailchimp"
   @import_failed_tag %{name: "Dispatch Import Failed", status: "active"}
 
-  def sync_existing_rider(%Rider{} = rider) do
-    with {:ok, account} <- Mailchimp.Account.get(),
-         {:ok, list} <- Mailchimp.Account.get_list(account, get_config(:list_id)),
-         {:ok, member} <- Mailchimp.List.get_member(list, rider.email),
-         {:ok, rider_attrs} <- build_rider(member) do
-      Riders.update_rider(rider, rider_attrs)
-    end
-  end
-
-  def sync_riders do
-    Multi.new()
-    |> Multi.run(:get_last_synced, fn repo, _changes ->
+  # TODO make this a singleton? / add locking
+  def sync_riders(opts \\ nil) do
+    Repo.transaction(fn ->
       last_synced =
-        case repo.get_by(Importer, name: @mailchimp) do
+        # TODO add a row lock here.
+        case Repo.get_by(Importer, name: @mailchimp) do
           nil -> nil
           importer -> importer.data["last_synced"]
         end
```

Comment:
```suggestion
 Repo.one(
          from i in Importer,
            where: i.name == ^"mailchimp",
            lock: "FOR UPDATE SKIP LOCKED",
            select: fragment("? ->> 'last_synced'", i.data)
        )
```

This is what the row lock looks like - it will cause all other transactions to get back nil while the transaction is running. we are locking this particular row of the importer tablw, which corresponds to just the mailchimp importer.

A good refactor would be to put that in the Importers module but I think this is the only one that needs it so far so fine to wait.